GPU/D3D11/FramebufferManagerD3D11.cpp

```diff
@@ -762,93 +749,44 @@ void FramebufferManagerD3D11::BlitFramebuffer(VirtualFramebuffer *dst, int dstX,
 		false);
 }
 
-// TODO: SSE/NEON
-// Could also make C fake-simd for 64-bit, two 8888 pixels fit in a register :)
-void ConvertFromRGBA8888(u8 *dst, u8 *src, u32 dstStride, u32 srcStride, u32 width, u32 height, GEBufferFormat format) {
-	// Must skip stride in the cases below.  Some games pack data into the cracks, like MotoGP.
-	const u32 *src32 = (const u32 *)src;
-
-	if (format == GE_FORMAT_8888) {
-		u32 *dst32 = (u32 *)dst;
-		if (src == dst) {
-			return;
-		} else {
-			for (u32 y = 0; y < height; ++y) {
-				memcpy(dst32, src32, width * 4);
-				src32 += srcStride;
-				dst32 += dstStride;
-			}
-		}
-	} else {
-		// But here it shouldn't matter if they do intersect
-		u16 *dst16 = (u16 *)dst;
-		switch (format) {
-		case GE_FORMAT_565: // BGR 565
-			for (u32 y = 0; y < height; ++y) {
-				ConvertRGBA8888ToRGB565(dst16, src32, width);
-				src32 += srcStride;
-				dst16 += dstStride;
-			}
-			break;
-		case GE_FORMAT_5551: // ABGR 1555
-			for (u32 y = 0; y < height; ++y) {
-				ConvertRGBA8888ToRGBA5551(dst16, src32, width);
-				src32 += srcStride;
-				dst16 += dstStride;
-			}
-			break;
-		case GE_FORMAT_4444: // ABGR 4444
-			for (u32 y = 0; y < height; ++y) {
-				ConvertRGBA8888ToRGBA4444(dst16, src32, width);
-				src32 += srcStride;
-				dst16 += dstStride;
-			}
-			break;
-		case GE_FORMAT_8888:
-		case GE_FORMAT_INVALID:
-			// Not possible.
-			break;
-		}
+static Draw::DataFormat GEFormatToThin3D(int geFormat) {
+	switch (geFormat) {
+	case GE_FORMAT_4444:
+		return Draw::DataFormat::A4R4G4B4_UNORM_PACK16;
+	case GE_FORMAT_5551:
+		return Draw::DataFormat::A1R5G5B5_UNORM_PACK16;
+	case GE_FORMAT_565:
+		return Draw::DataFormat::R5G6B5_UNORM_PACK16;
+	case GE_FORMAT_8888:
+		return Draw::DataFormat::R8G8B8A8_UNORM;
+	default:
+		return Draw::DataFormat::UNDEFINED;
 	}
 }
 
 // This function takes an already correctly-sized framebuffer and packs it into RAM.
 // Does not need to account for scaling.
 // Color conversion is currently done on CPU but should be done on GPU.
-void FramebufferManagerD3D11::PackFramebufferD3D11_(VirtualFramebuffer *vfb, int x, int y, int w, int h) {
+void FramebufferManagerD3D11::PackFramebufferSync_(VirtualFramebuffer *vfb, int x, int y, int w, int h) {
 	if (!vfb->fbo) {
 		ERROR_LOG_REPORT_ONCE(vfbfbozero, SCEGE, "PackFramebufferD3D11_: vfb->fbo == 0");
 		return;
 	}
 
 	const u32 fb_address = (0x04000000) | vfb->fb_address;
-	const int dstBpp = vfb->format == GE_FORMAT_8888 ? 4 : 2;
+
+	Draw::DataFormat destFormat = GEFormatToThin3D(vfb->format);
+	const int dstBpp = (int)DataFormatSizeInBytes(destFormat);
+
+	// TODO: Handle the other formats?  We don't currently create them, I think.
```

Comment:
This can probably be removed.  I think there's now a comment about the hardcoded 4 byte width of pixels.

-[Unknown]